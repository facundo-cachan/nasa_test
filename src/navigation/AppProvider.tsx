import React from 'react';
import styleLight from '../styles/light';
import styleDark from '../styles/dark';

export const AppContext: React.Context<any> = React.createContext(null);

export const AppProvider = ({children}: any) => {
  const [selectedStyles, setStyle] = React.useState(false),
    [day, setDay] = React.useState('2004-01-14');
  return (
    <AppContext.Provider
      value={{
        styles: selectedStyles ? styleLight : styleDark,
        day,
        setDay,
        setStyle,
        selectedStyles,
      }}>
      {children}
    </AppContext.Provider>
  );
};
