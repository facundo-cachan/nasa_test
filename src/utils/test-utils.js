import React from 'react'
import { render } from '@testing-library/react-native';
import { AppProvider } from '@navigation/AppProvider';

const AllTheProviders = ({ children }) => {
    return <AppProvider>{children}</AppProvider>;
};

const customRender = (ui, options) =>
    render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
