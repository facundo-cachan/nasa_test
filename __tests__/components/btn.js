import React from 'react'
import { fireEvent, render, waitFor } from 'test-utils'
import { AppContext } from '@navigation/AppProvider';

import { Btn } from '@components'

const BtnTest = () => {
  const { styles } = React.useContext(AppContext),
    btnStyle = {
      buttonContainer: styles.buttonContainer,
      iconWrapper: styles.iconWrapper,
      btnTxtWrapper: styles.btnTxtWrapper,
      buttonText: styles.buttonText,
    };
  return <Btn label="Test Button" icon="camera" styles={btnStyle} backgroundColor="#58D3F7" onPress={() => {
    setTimeout(() => {
      return true
    }, Math.floor(Math.random() * 200))
  }} />
}


describe('Testing components', () => {
  test('Btn test', async () => {
    const { getByTestId, toJSON, queryByTestId } = render(<BtnTest />);
    const button = getByTestId('btn01');
    fireEvent.press(button);
    await waitFor(() => expect(queryByTestId('btn01')).toBeTruthy());
    expect(toJSON()).toMatchSnapshot()
  })
})

