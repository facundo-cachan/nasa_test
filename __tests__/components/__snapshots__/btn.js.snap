// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing components Btn test 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#58D3F7",
      "borderRadius": 3,
      "flexDirection": "row",
      "height": 88.93333333333334,
      "justifyContent": "space-between",
      "margin": 10,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="btn01"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "width": 30,
      }
    }
  >
    <Text
      allowFontScaling={false}
      onPress={null}
      style={
        Array [
          Object {
            "color": "#F7D358",
            "fontSize": 22,
          },
          Object {},
          Object {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "900",
          },
        ]
      }
      type="FontAwesome5"
    >
      ï€°
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 18,
            "fontWeight": "bold",
          },
          Object {
            "color": "#BCA9F5",
          },
        ]
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;
